### PR TITLE
Teradata2

### PR DESCRIPTION
**Pull request by drepshas**
HHH-8191 Teradata 14.0

### CODE REVIEW

**Comment by brmeyer** [comment]
@drepshas, alright, many apologies for the absurd delay -- this really fell through the cracks.  One quick question:

I started to merge this, but noticed Dialect#isLockAppended.  It appears that's only applied to Select and SimpleSelect.  But there are other uses of Dialect#getForUpdateString: ForUpdateFragment, the ID generators, etc.  Should that be applied globally?

File: hibernate-core/src/test/java/org/hibernate/test/legacy/FooBarTest.java [hunk 1]
```diff
@@ -615,7 +616,7 @@ public void testQuery() throws Exception {
 		s.save(bar);
 		s.save(baz);
 		list = s.createQuery(
-				" from Bar bar where bar.baz.count=667 and bar.baz.count!=123 and not bar.baz.name='1-E-1'"
+				" from Bar bar where bar.baz.count=667 and bar.baz.count<>123 and not bar.baz.name='1-E-1'"
```
Comment (brmeyer) [inline hibernate-core/src/test/java/org/hibernate/test/legacy/FooBarTest.java]: This one might be an issue.  If I remember correctly, "<>" isn't supported by several DBs.  May need to revert this fix and just skip the test.

Comment (drepshas) [inline hibernate-core/src/test/java/org/hibernate/test/legacy/FooBarTest.java]: TeradataDialect#getForUpdateString returns an empty string. Also TeradataDialect#applyLocksToSql prepends the ForUpdateFragment to the SQL. This overrides the Dialect#applyLocksToSql which appends the ForUpdateFragment to the SQL. 
So, I think its ok as is. At least, it was enough to get me through the test suite. Let me know if you disagree.

Regarding the "<>",  I'm ok with skipping the test.

**Comment by brmeyer** [comment]
Manually merging into master (5.0)

**Pull request closed without merging**